CMakeLists.txt

```diff
@@ -0,0 +1,286 @@
+    )
+  else()
+    set(PYTHON_EXTENSION_SUFFIX
+      ".${PYTHON_SOABI}-${PYTHON_MULTIARCH}"
+      CACHE INTERNAL
+      "The full suffix for Python native extensions. See PEP-3149: https://www.python.org/dev/peps/pep-3149/."
+    )
+  endif()
+endif()
+
+if(WIN32)
+  set(PYTHON_EXTENSION_EXTENSION ".pyd")
+else()
+# Also use .so for OSX, not dylib
+  set(PYTHON_EXTENSION_EXTENSION ".so")
+endif()
+
+macro(target)
+  if(NOT "${target_suffix} " STREQUAL " ")
+    get_rcl_information("${rmw_implementation}" "rcl${target_suffix}")
+  endif()
+  if("${rmw_implementation} " STREQUAL "rmw_connext_dynamic_cpp ")
+    message(STATUS "Skipping tests for '${rmw_implementation}'")
+    return()
+  endif()
+  if("${rmw_implementation} " STREQUAL "rmw_fastrtps_cpp ")
+    message(STATUS "Skipping tests for '${rmw_implementation}'")
+    return()
+  endif()
+
+  add_library(
+    ${PROJECT_NAME}${target_suffix}
+    SHARED src/rclpy/_rclpy.c
+  )
+  target_compile_definitions(${PROJECT_NAME}${target_suffix}
+    PRIVATE "RMW_IMPLEMENTATION_SUFFIX=${target_suffix}")
+
+  set_target_properties(${PROJECT_NAME}${target_suffix} PROPERTIES
+    PREFIX ""
+    LIBRARY_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}"
+    RUNTIME_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}"
+    OUTPUT_NAME "_${PROJECT_NAME}${target_suffix}${PYTHON_EXTENSION_SUFFIX}"
+    SUFFIX "${PYTHON_EXTENSION_EXTENSION}")
+
+  set_target_properties(${PROJECT_NAME}${target_suffix} PROPERTIES
+    PREFIX ""
+    LIBRARY_OUTPUT_DIRECTORY_DEBUG "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}"
+    RUNTIME_OUTPUT_DIRECTORY_DEBUG "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}"
+    OUTPUT_NAME "_${PROJECT_NAME}${target_suffix}${PYTHON_EXTENSION_SUFFIX}"
+    SUFFIX "${PYTHON_EXTENSION_EXTENSION}")
+
+  set_target_properties(${PROJECT_NAME}${target_suffix} PROPERTIES
+    PREFIX ""
+    LIBRARY_OUTPUT_DIRECTORY_RELEASE "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}"
+    RUNTIME_OUTPUT_DIRECTORY_RELEASE "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}"
+    OUTPUT_NAME "_${PROJECT_NAME}${target_suffix}${PYTHON_EXTENSION_SUFFIX}"
+    SUFFIX "${PYTHON_EXTENSION_EXTENSION}")
+
+  set_target_properties(${PROJECT_NAME}${target_suffix} PROPERTIES
+    PREFIX ""
+    LIBRARY_OUTPUT_DIRECTORY_RELWITHDEBINFO "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}"
+    RUNTIME_OUTPUT_DIRECTORY_RELWITHDEBINFO "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}"
+    OUTPUT_NAME "_${PROJECT_NAME}${target_suffix}${PYTHON_EXTENSION_SUFFIX}"
+    SUFFIX "${PYTHON_EXTENSION_EXTENSION}")
+
+  set_target_properties(${PROJECT_NAME}${target_suffix} PROPERTIES
+    PREFIX ""
+    LIBRARY_OUTPUT_DIRECTORY_MINSIZEREL "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}"
+    RUNTIME_OUTPUT_DIRECTORY_MINSIZEREL "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}"
+    OUTPUT_NAME "_${PROJECT_NAME}${target_suffix}${PYTHON_EXTENSION_SUFFIX}"
+    SUFFIX "${PYTHON_EXTENSION_EXTENSION}")
```

Comment:
The five `set_target_properties` calls are almost identical and should be refactored into a function to avoid code duplication.